app/javascript/crayons/MarkdownToolbar/markdownSyntaxFormatters.jsx

```diff
@@ -714,6 +820,70 @@ export const coreSyntaxFormatters = {
         blockSuffix: '\n```',
       }),
   },
+  embed: {
+    icon: () => <Icon src={EmbedIcon} />,
+    label: 'Embed',
+    getKeyboardShortcut: () => {
+      const modifier = getOSKeyboardModifierKeyString();
+      return {
+        command: `${modifier}+shift+k`,
+        tooltipHint: `${modifier.toUpperCase()} + SHIFT + K`,
+      };
+    },
+    getFormatting: ({ selectionStart, selectionEnd, value }) => {
```

Comment:
I think now that the requirement to conditionally add `https://...` has been removed we could actually use the existing helper:

```
getFormatting: ({ selectionStart, selectionEnd, value }) =>
      undoOrAddFormattingForInlineSyntax({
        value,
        selectionStart,
        selectionEnd,
        prefix: '{% embed ',
        suffix: ' %}',
      }),
```
It would remove the need for the additional helper functions 🙂 
I tested locally, and all the test cases still passed, but see what you think!

Reply:
Thanks for simplifying this.